torch/_decomp/decompositions.py

```diff
@@ -1200,6 +1200,28 @@ def split(self: Tensor, split_size: int, dim: int = 0) -> Tuple[Tensor, ...]:
     return torch.split(self, split_sizes, dim)
 
 
+@aten.tensor_split.tensor_indices_or_sections.py_impl(
+    DispatchKey.CompositeImplicitAutograd
+)
+def tensor_split_tensor_indices_or_sections_py_impl(
+    self: Tensor,
+    tensor_indices_or_sections: Tensor,
+    dim: int = 0,
+) -> List[Tensor]:
+    split_dim = tensor_indices_or_sections.dim()
+    assert split_dim == 1 or split_dim == 0, (
```

Comment:
nit: f strings will format eagerly, even if the assertion passes. we use `torch._check` to lazily format exception messages